casper/src/main/scala/coop/rchain/casper/util/BondingUtil.scala

```diff
@@ -115,10 +121,15 @@ object BondingUtil {
     for {
       transferSigData <- walletTransferSigData[F](nonce, amount, destination)
       transferSig     = Secp256k1.sign(transferSigData, secKey)
-    } yield s"""
-             |for(wallet <- @"$pubKey") {
-             |  wallet!("transfer", $amount, $nonce, "${Base16.encode(transferSig)}", "$destination", "$transferStatusOut")
-             |}""".stripMargin
+    } yield s"""new rl(`rho:registry:lookup`), WalletCheckCh, result in {
+               |  rl!(`rho:id:oqez475nmxx9ktciscbhps18wnmnwtm6egziohc3rkdzekkmsrpuyt`, *WalletCheckCh) |
+               |  for(@(_, WalletCheck) <- WalletCheckCh) {
+               |    @WalletCheck!("access", "$pubKey", *result) |
+               |    for(@(true, wallet) <- result) {
```

Comment:
Does this `true` match anything? It is amazing that it just works.

Reply:
`WalletCheck` returns a pair where the first element is true/false (success/failure) and the second element is an error message or a wallet. I only want to continue on the success case. It would probably be good to give some indication to the user in the failure case, but there is not really a good way of doing that right now.